public/requests/queries/dashboard_queries.ts

```diff
@@ -31,39 +31,34 @@ export const getDashboardQuery = () => {
           size: 10000,
         },
         aggs: {
-          traces: {
-            terms: {
-              field: 'traceId',
-              order: {
-                last_updated: 'desc',
-              },
-              size: 10000,
-            },
-            aggs: {
-              duration: {
-                max: {
-                  field: 'traceGroupFields.durationInNanos',
-                },
-              },
-              last_updated: {
-                max: {
-                  field: 'traceGroupFields.endTime',
-                },
-              },
-            },
-          },
-          average_latency_nanos: {
-            avg_bucket: {
-              buckets_path: 'traces>duration',
-            },
-          },
           average_latency: {
-            bucket_script: {
-              buckets_path: {
-                count: '_count',
-                latency: 'average_latency_nanos.value',
-              },
-              script: 'Math.round(params.latency / 10000) / 100.0',
+            scripted_metric: {
+              init_script: 'state.map = [:]; state.total_latency = 0.0; state.count = 0.0;',
+              map_script: `
+                if (doc.containsKey('traceGroupFields.durationInNanos') && !doc['traceGroupFields.durationInNanos'].empty) {
+                  def id = doc['traceId'].value;
+                  if (!state.map.containsKey(id)) {
```

Comment:
I'm not sure if the `state.map` is a single map or exists per shard. My concern is if a trace were split across 2 shards, and if `state.map` is per shard, then the trace would be represented twice.